flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamOneInputProcessor.java

```diff
@@ -45,16 +45,16 @@
 	private final StreamTaskInput<IN> input;
 	private final DataOutput<IN> output;
 
-	private final OperatorChain<?, ?> operatorChain;
+	private final BoundedMultiInput endOfInputAware;
 
 	public StreamOneInputProcessor(
 			StreamTaskInput<IN> input,
 			DataOutput<IN> output,
-			OperatorChain<?, ?> operatorChain) {
+			BoundedMultiInput endOfInputAware) {
```

Comment:
nit: it would be nice to keep the remaining processors in sync with this change?

Reply:
`StreamMultipleInputProcessor` needs the `OperatorChain` for extracting chained sources. I updated it in the `StreamTwoInputProcessor` though.